src/main/java/ch/njol/skript/expressions/ExprPlayerTime.java

```diff
@@ -0,0 +1,133 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.expressions.base.SimplePropertyExpression;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.util.Time;
+import ch.njol.skript.util.Timeperiod;
+import ch.njol.skript.util.Timespan;
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+import org.bukkit.entity.Player;
+import org.bukkit.event.Event;
+import org.eclipse.jdt.annotation.Nullable;
+
+@Name("Player Time")
+@Description({
+	"Gets the player's time. This will set the player's own personal time.",
+	"When setting relative time for a player, their time will tick inline with the world, without it will be static."
+})
+@Examples({
+	"set player time of player to dusk, 10:00 or midnight # random out of the three options",
+	"add 10 hours to player's player time",
+	"remove 10:00 from player's player time",
+	"reset player time of players"
+})
+@Since("INSERT VERSION")
+public class ExprPlayerTime extends SimplePropertyExpression<Player, Time> {
+
+	static {
+		registerDefault(ExprPlayerTime.class, Time.class, "[:relative] player time", "players");
+	}
+
+	private boolean relative;
+
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		relative = parseResult.hasTag("relative");
+		return super.init(exprs, matchedPattern, isDelayed, parseResult);
+	}
+
+	@Override
+	public @Nullable Time convert(Player player) {
+		return new Time((int) player.getPlayerTime());
+	}
+
+	@Override
+	@Nullable
+	public Class<?>[] acceptChange(ChangeMode mode) {
+		switch (mode) {
```

Comment:
missing default, return null in the default so a return null isn't needed after switch.